src/authInfo.ts

```diff
@@ -603,6 +659,44 @@ export class AuthInfo extends AsyncCreatable<AuthInfo.Options> {
     return sfdxAuthUrl;
   }
 
+  /**
+   * Set the defaultusername or the defaultdevhubusername to the alias if
+   * it exists otherwise to the username. Method will try to set the local
+   * config first but will default to global config if that fails.
+   * @param options
+   */
+  public async setAsDefault(options: { defaultUsername?: boolean; defaultDevhubUsername?: boolean }) {
+    let config: Config;
+    // if we fail to create the local config, default to the global config
+    try {
+      config = await Config.create({ isGlobal: false });
+    } catch {
+      config = await Config.create({ isGlobal: true });
+    }
+
+    const username = ensureString(this.getUsername());
+    const aliases = await Aliases.create(Aliases.getDefaultOptions());
+    const value = aliases.getKeysByValue(username)[0] || username;
+
+    if (options.defaultUsername) {
+      config.set(Config.DEFAULT_USERNAME, value);
+    }
+
+    if (options.defaultDevhubUsername) {
+      config.set(Config.DEFAULT_DEV_HUB_USERNAME, value);
+    }
+    await config.write();
+  }
+
+  /**
+   * Sets the provided alias to the username
+   * @param alias alias to set
+   */
+  public async setAlias(alias: string) {
+    const username = this.getUsername();
+    await Aliases.parseAndUpdate([`${alias}=${username}`]);
```

Comment:
You could add a new method called `Aliases.update(alias: string, value: string)` if you want. This method was mainly to help with command line arguments but I'm also fine if you just leave it like this.